### DISCUSSION TITLE
Unable to Disable Editing/Interaction with Objects on Canvas

### DISCUSSION THREAD

**Discussion by anan1213095357** [fabricjs/fabric.js#9439] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

4.5.0

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

18.16.0

### Link To Reproduction

null

### Steps To Reproduce

I am trying to switch between two modes in my application - an 'Edit Mode' and a 'Render Mode'. In 'Edit Mode', the user should be able to interact with, edit, and move the objects on the canvas. However, in 'Render Mode', all interaction should be disabled and the canvas should only display the objects without allowing any user interaction.

I have tried using canvas.selection = false; and setting object.selectable = false; for each object on the canvas, but the objects can still be moved, scaled, and rotated. I have also attempted to use canvas.interactive = false; but it didn't resolve the issue.

Here's a snippet of the code I am using:

`export function switchToRenderMode() {
    canvas.selection = false;
    canvas.forEachObject(function (object) {
        object.selectable = false;
    });
    canvas.renderAll();
}
`

Is there a built-in method or a recommended way to disable all interactions with the objects on the canvas while in 'Render Mode'? Any help would be appreciated!

This draft outlines the problem you're facing, the attempts you've made to resolve it, and a request for a solution or recommendation. Feel free to modify any part of this draft to better fit your situation or to add any additional information you think may be helpful.

### Expected Behavior

Refer to above

### Actual Behavior

Refer to above

**Comment by ShaMan123** [comment, marked as answer]
I agree we should have a preventDefault machnism
Meanwhile try `canvas.skipTargetFind = true` - that will disable finding objects under the cursor

**Comment by asturur** [comment]
Also setting the objects to obj.evented = false should do the job ok
